### PR TITLE
provide better handling of network error/user inputs

### LINKED ISSUE #438: [BUG] Provide better handling of network errors
**Description**
When connection to UTBot is not stable CLion shows unclear messages.

**To Reproduce**
1. Restart UTBot server with `utbot_server_restart.sh`
2. During restarting try to generate tests in CLion - as result unclear message shown
image
3. When CLion client reconnected message `Project configured` shown
image

**Expected**
1. When connection lost - balloon with error appears which describes the problem
2. When connection lost - info balloon notifies that connection restored. 

**Errors in Client log console and idea.log**
```
10:28:40.0509 | Client.kt:177 [DefaultDispatcher-worker-2] |ERROR| Heartbeat failed with grpc io exception 
UNAVAILABLE: io exception 
10:28:40.0529 | Client.kt:177 [DefaultDispatcher-worker-6] |ERROR| Exception when executing server request 
UNAVAILABLE: io exception 
```

```
2022-09-07 10:18:04,928 [ 486068]   INFO - STDERR - io.grpc.StatusException: UNAVAILABLE: io exception
2022-09-07 10:18:04,928 [ 486068]   INFO - STDERR - 	at io.grpc.Status.asException(Status.java:551)
2022-09-07 10:18:04,928 [ 486068]   INFO - STDERR - 	at io.grpc.kotlin.ClientCalls$rpcImpl$1$1$1.onClose(ClientCalls.kt:295)
2022-09-07 10:18:04,928 [ 486068]   INFO - STDERR - 	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:557)
2022-09-07 10:18:04,928 [ 486068]   INFO - STDERR - 	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:69)
2022-09-07 10:18:04,928 [ 486068]   INFO - STDERR - 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:738)
2022-09-07 10:18:04,928 [ 486068]   INFO - STDERR - 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:717)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at java.base/java.lang.Thread.run(Thread.java:833)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - Caused by: java.net.SocketException: Connection reset
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:394)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:426)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:253)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1133)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:350)
2022-09-07 10:18:04,929 [ 486069]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151)
2022-09-07 10:18:04,930 [ 486070]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
2022-09-07 10:18:04,930 [ 486070]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
2022-09-07 10:18:04,930 [ 486070]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
2022-09-07 10:18:04,930 [ 486070]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
2022-09-07 10:18:04,930 [ 486070]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
2022-09-07 10:18:04,930 [ 486070]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2022-09-07 10:18:04,930 [ 486070]   INFO - STDERR - 	at io.grpc.netty.shaded.io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2022-09-07 10:18:04,930 [ 486070]   INFO - STDERR - 	... 1 more
```

### LINKED ISSUE #444: [BUG] UTBot doesn't properly handle wrong setting of Path to project on remote machine setting
**To reproduce**
1. In the settings of UTBot set `/` as Path to project on remote machine
image

2. In UTBot menu invoke 'Configure project' - as result exception thrown:

```
java.nio.file.InvalidPathException: UNC path is missing sharename: /\tests
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:118)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:232)
	at java.base/java.nio.file.Path.of(Path.java:147)
	at java.base/java.nio.file.Paths.get(Paths.java:69)
	at org.utbot.cpp.clion.plugin.utils.PathUtilsKt.convertToRemotePath(PathUtils.kt:118)
	at org.utbot.cpp.clion.plugin.utils.PathUtilsKt.convertToRemotePathIfNeeded(PathUtils.kt:100)
	at org.utbot.cpp.clion.plugin.settings.UTBotAllProjectSettings.getConvertedTestDirPath(UTBotAllProjectSettings.kt:32)
	at org.utbot.cpp.clion.plugin.grpc.GrpcMessagingUtilsKt.getProjectContextMessage(GrpcMessagingUtils.kt:23)
	at org.utbot.cpp.clion.plugin.grpc.ConfigurationGrpcRequestsKt.getProjectConfigGrpcRequest(ConfigurationGrpcRequests.kt:14)
	at org.utbot.cpp.clion.plugin.client.requests.CheckProjectConfigurationRequest.<init>(CheckProjectConfigurationRequest.kt:19)
	at org.utbot.cpp.clion.plugin.actions.configure.ConfigureProjectAction.actionPerformed(ConfigureProjectAction.kt:11)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.doPerformActionOrShowPopup(ActionUtil.java:315)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performActionDumbAwareWithCallbacks$4(ActionUtil.java:294)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:337)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAwareWithCallbacks(ActionUtil.java:294)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.invokeAction(ActionUtil.java:516)
	at com.intellij.ui.popup.ActionPopupStep.performAction(ActionPopupStep.java:232)
	at com.intellij.ui.popup.ActionPopupStep.lambda$onChosen$1(ActionPopupStep.java:220)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:94)
	at com.intellij.ui.popup.AbstractPopup.lambda$dispose$18(AbstractPopup.java:1543)
	at com.intellij.util.ui.EdtInvocationManager.invokeLaterIfNeeded(EdtInvocationManager.java:113)
	at com.intellij.ide.IdeEventQueue.ifFocusEventsInTheQueue(IdeEventQueue.java:180)
	at com.intellij.ide.IdeEventQueue.executeWhenAllFocusEventsLeftTheQueue(IdeEventQueue.java:133)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:164)
	at com.intellij.ui.popup.AbstractPopup.dispose(AbstractPopup.java:1540)
	at com.intellij.ui.popup.WizardPopup.dispose(WizardPopup.java:162)
	at com.intellij.ui.popup.list.ListPopupImpl.dispose(ListPopupImpl.java:326)
	at com.intellij.ui.popup.PopupFactoryImpl$ActionGroupPopup.dispose(PopupFactoryImpl.java:266)
	at com.intellij.openapi.util.ObjectTree.runWithTrace(ObjectTree.java:126)
	at com.intellij.openapi.util.ObjectTree.executeAll(ObjectTree.java:158)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:217)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:205)
	at com.intellij.ui.popup.WizardPopup.disposeAllParents(WizardPopup.java:266)
	at com.intellij.ui.popup.list.ListPopupImpl.handleNextStep(ListPopupImpl.java:434)
	at com.intellij.ui.popup.list.ListPopupImpl._handleSelect(ListPopupImpl.java:406)
	at com.intellij.ui.popup.list.ListPopupImpl.handleSelect(ListPopupImpl.java:361)
	at com.intellij.ui.popup.PopupFactoryImpl$ActionGroupPopup.handleSelect(PopupFactoryImpl.java:278)
	at com.intellij.ui.popup.list.ListPopupImpl$MyMouseListener.mouseReleased(ListPopupImpl.java:618)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:298)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6648)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3392)
	at com.intellij.ui.popup.list.ListPopupImpl$MyList.processMouseEvent(ListPopupImpl.java:694)
	at java.desktop/java.awt.Component.processEvent(Component.java:6413)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5022)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2802)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:754)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:752)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:751)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:918)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:840)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:450)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:791)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:449)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:624)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:447)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:881)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:493)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

**Expected**
Whatever it can CLion should check settings right away in UTBot setting and not allow to save invalid setting.
If user still entered wrong value of this setting UTBot should show common error message with human readable description of the problem.

### PR DESCRIPTION
**Pull request by vol0n**
closes #444, closes #438

- better user notifications with titles. 
- More grpc errors are handled
- now exceptions related to bad paths are caught 
- added validation to user input in UI

### CODE REVIEW

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/ui/services/OutputProvider.kt [hunk 1]
```diff
@@ -65,4 +65,8 @@ class OutputProvider(val project: Project) : Disposable {
             }
         }
     }
+
+    companion object {
```
Comment (alexey-utkin) [inline clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/ui/services/OutputProvider.kt:69]: Do you need this object?
Do you use it for sync?

Comment (vol0n) [reply clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/ui/services/OutputProvider.kt]: fixed

**Review by alexey-utkin — APPROVED** [review]

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/ui/UTBotRequestProgressIndicator.kt [hunk 2]
```diff
@@ -32,15 +33,21 @@ class UTBotRequestProgressIndicator(
     }
 
     fun stopShowingProgressInUI() {
-        finish()
-        super.stop()
+        // stop can be called only if progress was started (=running)
+        if (isRunning) {
+            finish()
+            super.stop()
```
Comment (alexey-utkin) [inline clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/ui/UTBotRequestProgressIndicator.kt:39]: The call of `super` notation outside overwritten method looks suspicious.

Comment (vol0n) [reply clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/ui/UTBotRequestProgressIndicator.kt]: fixed: now I make sure the super.start is called in start(), and do not check for running status

**Pull request merged**